src/streamlink/plugins/vimeo.py

```diff
@@ -8,70 +8,176 @@
 import logging
 import re
 from html import unescape as html_unescape
+from time import time
 from urllib.parse import urlparse
 
+from streamlink.exceptions import StreamError
 from streamlink.plugin import Plugin, PluginArgument, PluginArguments, pluginmatcher
 from streamlink.plugin.api import validate
 from streamlink.stream.dash import DASHStream
 from streamlink.stream.ffmpegmux import MuxedStream
-from streamlink.stream.hls import HLSStream
+from streamlink.stream.hls import HLSStream, MuxedHLSStream
 from streamlink.stream.http import HTTPStream
 
 log = logging.getLogger(__name__)
 
 
-@pluginmatcher(re.compile(
-    r"https?://(player\.vimeo\.com/video/\d+|(www\.)?vimeo\.com/.+)"
-))
-class Vimeo(Plugin):
-    _config_url_re = re.compile(r'(?:"config_url"|\bdata-config-url)\s*[:=]\s*(".+?")')
-    _config_re = re.compile(r"var\s+config\s*=\s*({.+?})\s*;")
-    _config_url_schema = validate.Schema(
-        validate.transform(_config_url_re.search),
-        validate.any(
-            None,
-            validate.Schema(
-                validate.get(1),
-                validate.parse_json(),
-                validate.transform(html_unescape),
-                validate.url(),
-            ),
-        ),
-    )
+class VimeoMuxedHLSStream(MuxedHLSStream):
+    def __init__(self, session, video, audio, **args):
+        super().__init__(session, video, audio, hlsstream=VimeoHLSStream, **args)
+
+
+class VimeoHLSStream(HLSStream):
+    __muxed__ = VimeoMuxedHLSStream
+
+    _expiry_time_limit = 60
+
+    def __init__(self, session_, url, **args):
+        self.session = session_
+        self._url = url
+        self.api = args.pop("api")
+        self._parsed_url = urlparse(self._url)
+        self._path_parts = self._parsed_url.path.split("/")
+        self._need_update = False
+        self._set_expiry_time(self._path_parts[1])
+        super().__init__(self.session, self._url, **args)
+
+    def _set_expiry_time(self, path):
+        m = VimeoAPI._expiry_re1.search(path) or VimeoAPI._expiry_re2.search(path)
+        if not m:
+            raise StreamError("expiry value not found in URL")
+        self.api.expiry_time = int(m.group(1)) - self._expiry_time_limit
```

Comment:
This should be moved to the API class, as you're only accessing API class attributes here.